tgui/packages/tgui/interfaces/SecureSafe.js

```diff
@@ -0,0 +1,112 @@
+import { Box, Button, Divider, Flex, Section } from "../components";
+import { Window } from "../layouts";
+import { useBackend } from "../backend";
+
+const buttons = [[1, 2, 3], [4, 5, 6], [7, 8, 9], ['R', 0, 'E']];
+
+export const SecureSafe = (properties, context) => {
+  const { data, act } = useBackend(context);
+
+  const {
+    l_set,
+    l_code_len,
+    emagged,
+    locked,
+    code,
+    error,
+  } = data;
+
+  const onButtonClick = val => {
+    switch (val) {
+      case 'R':
+        act('reset');
+        break;
+      case 'E':
+        act('enter');
+        break;
+      default:
+        act('addCode', { number: val });
+        break;
+    }
+  };
+
+  let codeChars = (code || '').split('');
+  let codeBoxes = [];
+
+  for (let i = 0; i < l_code_len; i++) {
+    codeBoxes.push(
+      <Flex
+        className="SecureSafe__CodeBox"
+        alignItems="center"
+        justifyContent="center"
+        flexGrow={1}
+        key={i}>
+        {locked ? codeChars[i] || '_' : '*'}
+      </Flex>
+    );
+  }
+
+  let message = null;
+
+  if (emagged) {
+    message = 'LOCKING SYSTEM ERROR - 1701';
+  } else if (error) {
+    message = error;
+  } else if (!l_set) {
+    message = 'ENTER NEW PASSCODE';
+  }
+
+
+  return (
+    <Window>
+      <Window.Content>
+        <Section>
+
+          <Flex flexDirection="row" alignItems="center" justifyContent="space-between">
```

Comment:
@dearmochi check this out, your flex properties on box have single handedly made the Flex component obsolete.

Reply:
... Which I think is a good thing **personally** because I don't want to box things over and over for. It gets crowded really quickly when you have flex layouts inside other flex layouts

Reply:
One of the reasons Flex exists in its current opinionated form, is that it allows easy polyfilling of its functionality for IE8.

Reply:
Fair enough. I'll admit that I'm far from experienced with web development and polyfilling certainly didn't come across my mind when I added the `Box` properties. I can see how it makes sense to have a separate `Flex` component now.

Reply:
ie8 lol?